js/src/components/free-listings/setup-free-listings/index.js

```diff
@@ -222,10 +225,36 @@ const SetupFreeListings = ( {
 				validate={ handleValidate }
 				onSubmit={ onContinue }
 			>
-				<FormContent submitLabel={ submitLabel } />
+				{ ( formContext ) => {
+					const { isValidForm, handleSubmit, adapter } = formContext;
+					const handleSubmitClick = ( event ) => {
+						if ( isValidForm ) {
+							return handleSubmit( event );
+						}
+
+						adapter.showValidation();
+					};
+
+					return (
+						<>
+							<FormContent />
+							<Fill>
+								<AppButton
+									isPrimary
+									loading={ adapter.isSubmitting }
+									onClick={ handleSubmitClick }
+								>
+									{ submitLabel }
+								</AppButton>
+							</Fill>
+						</>
+					);
+				} }
 			</AdaptiveForm>
 		</div>
 	);
 };
 
+SetupFreeListings.SubmitButton = Slot;
+
```

Comment:
This part is not very easy to understand for me. I assume that when we use 
 <SetupFreeListings.SubmitButton /> (the Slot) sowhere. It will render whatever is inside <Fill>?

Reply:
Yes. The content inside the `<Fill>` will be rendered to the paired `<Slot>`, and the component consumer side decides where the slot is placed. In this way, the submit button can be rendered more flexibly outside of `SetupFreeListings` and can be composited with other components.

It's used here to present different layouts. One has stepper wrappers and one doesn't.